src/preamble.js

```diff
@@ -839,56 +832,51 @@ function instantiateAsync(binary, binaryFile, imports) {
-        // "response was already consumed" error.)
-        var clonedResponsePromise = response.clone().arrayBuffer();
+      // We need the wasm binary for the offset converter. Clone the response
+      // in order to get its arrayBuffer (cloning should be more efficient
+      // than doing another entire request).
+      // (We must clone the response now in order to use it later, as if we
+      // try to clone it asynchronously lower down then we will get a
+      // "response was already consumed" error.)
+      var clonedResponsePromise = response.clone().arrayBuffer();
 #endif
 
-        result.then(
 #if USE_OFFSET_CONVERTER
-          (instantiationResult) => {
-            // When using the offset converter, we must interpose here. First,
-            // the instantiation result must arrive (if it fails, the error
-            // handling later down will handle it). Once it arrives, we can
-            // initialize the offset converter. And only then is it valid to
-            // call receiveInstantiationResult, as that function will use the
-            // offset converter (in the case of pthreads, it will create the
-            // pthreads and send them the offsets along with the wasm instance).
-
-            clonedResponsePromise.then((arrayBufferResult) => {
-                wasmOffsetConverter = new WasmOffsetConverter(new Uint8Array(arrayBufferResult), instantiationResult.module);
-                resolve(instantiationResult);
-              },
-              (reason) => err(`failed to initialize offset-converter: ${reason}`)
-            );
-          },
-#else
-          resolve,
-#endif
-          (reason) => {
-            // We expect the most common failure cause to be a bad MIME type for the binary,
-            // in which case falling back to ArrayBuffer instantiation should work.
-            err(`wasm streaming compile failed: ${reason}`);
-            err('falling back to ArrayBuffer instantiation');
-            return resolve(instantiateArrayBuffer(binaryFile, imports));
-          }
-        );
-      });
-    });
+      var instantiationResult = await result;
+      // When using the offset converter, we must interpose here. First,
+      // the instantiation result must arrive (if it fails, the error
+      // handling later down will handle it). Once it arrives, we can
+      // initialize the offset converter. And only then is it valid to
+      // call receiveInstantiationResult, as that function will use the
+      // offset converter (in the case of pthreads, it will create the
+      // pthreads and send them the offsets along with the wasm instance).
+
+      var arrayBufferResult = await clonedResponsePromise;
+      try {
+        wasmOffsetConverter = new WasmOffsetConverter(new Uint8Array(arrayBufferResult), instantiationResult.module);
+          resolve(instantiationResult);
+      } catch (reason) {
+        (reason) => err(`failed to initialize offset-converter: ${reason}`)
```

Comment:
This should just call err.

Reply:
Should it also call abort? Before the promise rejected, but now IIUC the try-catch swallows the exception and we'd try to keep going.

Reply:
In the only code we use a promise rejection callback that just logs the error and does nothing else, so I think this is the same thing.

Reply:
When you have a rejection handler that rejection doesn't propagate out (IIUC).